plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestCachingDirectoryLister.java

```diff
@@ -60,6 +70,44 @@ protected QueryRunner createQueryRunner()
                 .build();
     }
 
+    private Table getTable(String schemaName, String tableName)
+    {
+        return fileHiveMetastore.getTable(schemaName, tableName)
+                .orElseThrow(() -> new NoSuchElementException(format("The table %s.%s could not be found", schemaName, tableName)));
+    }
+
+    private void sleep(long milliseconds) {
+        try {
+            Thread.sleep(milliseconds);
+        } catch (InterruptedException e) {
+            fail();
+        }
+    }
+
+    @Test(timeOut = 500)
+    public void testList_shouldNotTimeout() throws IOException {
+        sleep(500);
+
+        Configuration conf = new Configuration();
+        FileSystem fs = FileSystem.get(conf);
+        Table table = getTable(TPCH_SCHEMA,"partial_cache_invalidation_table1");
+        Path path = createTempDirectory(null);
+
+       cachingDirectoryLister.list(fs,table, (org.apache.hadoop.fs.Path) path);
+    }
+
+    @Test(timeOut = 1500)
+    public void testList_shouldTimeout() throws IOException {
```

Comment:
Great code, just add a comment on what it does. mention the default config of 1 minute timeout.